### ISSUE TITLE
Programmatic check for WDDM support

### ISSUE THREAD

**Issue by ghotik** [narzoul/DDrawCompat#413]
Hi again!
After the successful additions to DxWnd thanks to your support, since DxWnd is used also by a Windows XP community (at, least, whenever I make a change that breaks XP support there's soon someone who protests) I was wondering if there's a reliable way to detect the WDDM support. 
I temporarily used the OS version as a criteria since, for sure, Windows XP doesn't support WDDM. But in effect on Vista the support seems dependent on the release and in theory it could be missing also on even more recent OSs if the video driver doesn't support it, so I wanted to find a better criteria.
In DirectDrawCompat the issue seems absent since everything is based on WDDM support, but I wondered if you know some way to get the information programmatically, at least a boolean YES/NO about it.
Thanks
gho

**Comment by narzoul** [comment]
I don't know of a direct way to check.  I suspect the easiest option would be to just call DirectDrawCreate(Ex) and check if OpenAdapter gets called during that, or something like that. At least I assume the older driver model didn't use an OpenAdapter function, but I have no clue how it actually worked, never had to research that. It should still be documented by MS however.

**Comment by ghotik** [comment]
I kept trying to understand how to detect the support for WDDM, but the matter is complex. What I found is that apparently there is no programmatic support for the detection, but in addition the support is specific for each video adapter, so it could be available depending on the screen of a multi-screen configuration or the selection for a multi-card. It could also change in time after sw upgrades. So, in conclusion I think that a reasonable choice is to exclude WDDM on Windows XP (where it can't possibly be) and suppose it could be supported everywhere else.
Thank you again, I think I can close the topic now.
